packages/blade/src/components/ProgressBar/_decisions/decisions.md

```diff
@@ -0,0 +1,102 @@
+| size | `large`, `small` | `small` | The size of the progress bar to be rendered. |  |
+| accessibilityLabel | `string` | `"Loading"` | The accessibility label (aria-label) for the progress bar. |
+| isIndeterminate | `boolean` | `false` | Control whether the progress bar is indeterminate or not. The progress bar would be in an indeterminate state if `progress` is not passed.  |
+| showPercentage | `boolean` | `true` | Control whether the progress bar should show the current progress percentage or not. Percentage would be automatically hidden when the progress bar is in an indeterminate state.  |
+
+### Sample Usage
+```jsx
+import { ProgressBar } from '@razorpay/components';
+
+<ProgressBar 
+  label='Loading' 
+  value={20} 
+  contrast='low'
+  size='large'
+  accessibilityLabel='Loading Payments'
+/>
+```
+
+### Accessibility
+- We'll accept a prop `accessibilityLabel` which will be passed on as `aria-label` for web & `accessibilityLabel` for native to aide screen readers
+- By default the value of `accessibilityLabel` will be the same as `label` prop with a fallback to `"Loading"` and can be overwritten with something more meaningful by the consumer.
+- We will set `aria-role` as `progressbar`
+- We will pass the value of `value` prop as `aria-valuenow`
+
+## Handling pulsating animation
+After `3000ms` the progress bar would start pulsating to indicate to the user that the process is slow but isn't stuck.
+<img src="./progress-bar-pulse.gif" width="380" />
+
+**Problem:**
+- We have use-cases where a progress bar could be used:
+  - <img src="./progress-bar-static.png" width="240" />
+
+- In such cases, we wouldn't want the progress bar to pulsate after `3000ms` since this will remain static.
+
+**Possible Solutions:**
+1. Use prop `variant` as `'static'` and `'default' | 'animated'`
+
+- Pros:
+   - `variant` is a commonly used prop and makes it easy to understand that a different variation of the component exists
+
+- Cons:
+   - When the variant would be `static`, it won't entirely be static. When the `value` is changed, it will animate (easing with duration for filling the bar) to the new value. Only the pulsating animation would be disabled.
+   - Education effort would be needed to communicate exactly what we mean by a static variant and where should it be used.
+
+2. Use prop `animation` as `'none'` and `'default' | 'pulse'`
+
+- Pros:
+   - `animation` is more specific than `variant` since the only change would be related to animation.
+
+- Cons:
+  - When the animation would be `none`, it won't entirely be none. When the `value` is changed, it will animate (easing with duration for filling the bar) to the new value. Only the pulsating animation would be disabled.
+  - Education effort would be needed to communicate exactly what we mean by animation as none, pulse (or default) and where should it be used.
+  
+3. Use prop `showPulsatingAnimation` as `boolean`
```

Comment:
How about we use this approach with a more descriptive name?
- `animateAfterDelay`
- `showAnimationAfterDelay`
- `showWaitingAnimation`
- etc. naming things is hard 😅

Reply:
I thought about `animateAfterDelay` & `showAnimationAfterDelay` but it sounds misleading and might look like a prop that accepts a numeric "delay" after which some animation will start. "Pulsating" is more explicit about the kind of animation. I thought of using "Secondary" as `showSecondaryAnimation` but that seemed too vague. 

`showWaitingAnimation` seems okay to me

Reply:
`isAnimated` sounds good to me. As someone using a progress bar component, I just wanna control whether the progress bar is animated or not. "WaitingAnimation" sounds like the progress bar is built for a specific waiting context.

Reply:
>I just wanna control whether the progress bar is animated or not

Thats the catch with this prop. We're not going to entirely disable all animations if you pass `isAnimated=false`. We'll only be disabling the pulsing animation that starts after 3 seconds.